app/controllers/nonsipp/common/PartnershipRecipientUtrController.scala

```diff
@@ -78,11 +80,12 @@ class PartnershipRecipientUtrController @Inject()(
               .successful(BadRequest(view(formWithErrors, viewModel(srn, index, mode, subject, request.userAnswers)))),
           value =>
             for {
-              updatedAnswers <- Future
-                .fromTry(
-                  request.userAnswers.set(PartnershipRecipientUtrPage(srn, index, subject), ConditionalYesNo(value))
-                )
-              _ <- saveService.save(updatedAnswers)
+              updatedAnswers <- request.userAnswers
+                .set(PartnershipRecipientUtrPage(srn, index, subject), ConditionalYesNo(value))
+                .mapK
+              nextPage = navigator.nextPage(PartnershipRecipientUtrPage(srn, index, subject), mode, updatedAnswers)
+              updatedProgressAnswers <- saveProgress(srn, index, updatedAnswers, nextPage)
+              _ <- saveService.save(updatedProgressAnswers)
             } yield Redirect(navigator.nextPage(PartnershipRecipientUtrPage(srn, index, subject), mode, updatedAnswers))
```

Comment:
Same here. And any few other places further down in the PR where redirection happens.

Reply:
This has been addressed in a code cleanup , commit needs to yet be pushed